ImageProcessing/BrighterFatterCorrection.ipynb

```diff
@@ -0,0 +1,767 @@
+    "# Plot the brighter-fatter effect on those shape measurements and the corrected version\n",
+    "Alongside stamps of each object, below we show the trend of X and Y sizes before and after brighter-fatter correction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sz=11 # stamp size\n",
+    "\n",
+    "# loop over some objects and save a summary figure\n",
+    "# [0,6,12,18,23,35,44,46,52,56,69,71,114] are good indices to look with default values \n",
+    "# or e.g. np.random.choice(np.arange(nspots),size=10)\n",
+    "indexfoo=[0,6,12,18,23,35,44,46,52,56,69,71,114]\n",
+    "\n",
+    "for nfoo in indexfoo:\n",
+    "    plt.figure(figsize=(14,4)),plt.subplots_adjust(wspace=.3)\n",
+    "    \n",
+    "    # grab a postage stamp, integerizing the centroid and shipping\n",
+    "    # if it is near the edge, +TODO in a stackly manner\n",
+    "    xc,yc=bf_dat[10,nfoo,0].astype('int')+1,bf_dat[10,nfoo,1].astype('int')+1\n",
+    "    if ((np.abs(xc-250)>250 - sz ) | (np.abs(yc-250)>250 - sz )): continue\n",
+    "    stamp=exposure.getImage().array[yc-sz:yc+sz,xc-sz:xc+sz]\n",
+    "    \n",
+    "    # show the stamp with log scale (1,max)\n",
+    "    plt.subplot(131),plt.title('stamp '+str(nfoo).zfill(3)+' (log scale)')\n",
+    "    plt.imshow(stamp,origin='lower',norm=LogNorm(1,stamp.max())),plt.colorbar()\n",
+    "    \n",
+    "    # x size vs flux\n",
+    "    plt.subplot(132),plt.title('x (row) size vs. flux')\n",
+    "    plt.plot(bf_dat[:,nfoo,6],bf_dat[:,nfoo,2],'r.',label='Uncorrected')\n",
+    "    plt.plot(bf_dat[:,nfoo,7],bf_dat[:,nfoo,4],'g.',label='Corrected')\n",
+    "    plt.xlabel(flux_param),plt.ylabel(bf_param1),plt.xscale('log')\n",
+    "    plt.legend(loc='upper left')\n",
+    "\n",
+    "    # y size vs flux\n",
+    "    plt.subplot(133),plt.title('y (column) size vs. flux')\n",
+    "    plt.plot(bf_dat[:,nfoo,6],bf_dat[:,nfoo,3],'r.',label='Uncorrected')\n",
+    "    plt.plot(bf_dat[:,nfoo,7],bf_dat[:,nfoo,5],'g.',label='Corrected')\n",
+    "    plt.xlabel(flux_param),plt.ylabel(bf_param2)\n",
+    "    plt.xscale('log')\n",
+    "    plt.savefig(cat_dir+str(nfoo).zfill(5)+'bfcorr.png')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now plot the flux lost/gained in the process of brighter-fatter correction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "colorpalette=cycle(plt.cm.viridis(np.linspace(0,1,len(indexfoo))))\n",
+    "stylepalette=cycle(['s','*','o'])\n",
+    "plt.figure(figsize=(8,5))\n",
+    "for nfoo in indexfoo:\n",
+    "    flux_foo=bf_dat[:,nfoo,6]\n",
+    "    fluxdiffpct_foo=(bf_dat[:,nfoo,6]-bf_dat[:,nfoo,7])/bf_dat[:,nfoo,6]*100.\n",
+    "    plt.plot(flux_foo,fluxdiffpct_foo,label=str(nfoo).zfill(3),c=next(colorpalette),marker=next(stylepalette))\n",
```

Comment:
Would it be possible to make separate plots here for simulated stars and galaxies?